kong/clustering/data_plane.lua

```diff
@@ -139,13 +148,17 @@ local function send_ping(c, log_suffix)
 
   local _, err = c:send_ping(hash)
   if err then
+    set_control_plane_connected(false)
     ngx_log(is_timeout(err) and ngx_NOTICE or ngx_WARN, _log_prefix,
             "unable to send ping frame to control plane: ", err, log_suffix)
 
   -- only log a ping if the hash changed
-  elseif hash ~= prev_hash then
-    prev_hash = hash
-    ngx_log(ngx_INFO, _log_prefix, "sent ping frame to control plane with hash: ", hash, log_suffix)
+  else
+    set_control_plane_connected(true)
+    if hash ~= prev_hash then
```

Comment:
If we keep the code as is, I suggest moving the comment string on line 155 so that it continues to point to the correct context

```suggestion
    -- only log a ping if the hash changed
    if hash ~= prev_hash then
```